sermant-plugins/sermant-mq-grayscale/mq-grayscale-rocketmq-plugin/src/main/java/io/sermant/mq/grayscale/rocketmq/interceptor/RocketMqPullConsumerSubscriptionUpdateInterceptor.java

```diff
@@ -0,0 +1,69 @@
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package io.sermant.mq.grayscale.rocketmq.interceptor;
+
+import io.sermant.core.common.LoggerFactory;
+import io.sermant.core.plugin.agent.entity.ExecuteContext;
+import io.sermant.core.utils.ReflectUtils;
+import io.sermant.core.utils.StringUtils;
+import io.sermant.mq.grayscale.rocketmq.service.RocketMqConsumerGroupAutoCheck;
+import io.sermant.mq.grayscale.rocketmq.utils.RocketMqGrayscaleConfigUtils;
+import io.sermant.mq.grayscale.rocketmq.utils.RocketMqSubscriptionDataUtils;
+
+import org.apache.rocketmq.client.consumer.DefaultMQPullConsumer;
+import org.apache.rocketmq.client.impl.consumer.DefaultMQPullConsumerImpl;
+import org.apache.rocketmq.client.impl.factory.MQClientInstance;
+import org.apache.rocketmq.common.protocol.heartbeat.SubscriptionData;
+
+import java.util.Optional;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * update pull consumer subscription SQL92 query statement interceptor
+ *
+ * @author chengyouling
+ * @since 2024-07-27
+ **/
+public class RocketMqPullConsumerSubscriptionUpdateInterceptor extends RocketMqAbstractIntercepter {
+    private static final Logger LOGGER = LoggerFactory.getLogger();
+
+    @Override
+    public ExecuteContext doAfter(ExecuteContext context) throws Exception {
+        SubscriptionData subscriptionData = (SubscriptionData) context.getResult();
+        if (RocketMqSubscriptionDataUtils.isExpressionTypeInaccurate(subscriptionData.getExpressionType())) {
+            return context;
+        }
+        Optional<Object> fieldValue = ReflectUtils.getFieldValue(context.getObject(), "mQClientFactory");
+        if (!fieldValue.isPresent()) {
+            LOGGER.log(Level.SEVERE, "field mQClientFactory is not exist!");
+            return context;
+        }
+        DefaultMQPullConsumer pullConsumer
+                = ((DefaultMQPullConsumerImpl) context.getObject()).getDefaultMQPullConsumer();
+        String consumerGroup = pullConsumer.getConsumerGroup();
+        MQClientInstance instance = (MQClientInstance) fieldValue.get();
+        String namesrvAddr = instance.getClientConfig().getNamesrvAddr();
+        if (StringUtils.isEmpty(RocketMqGrayscaleConfigUtils.getGrayGroupTag())) {
+            RocketMqConsumerGroupAutoCheck.setMqClientInstance(subscriptionData.getTopic(), consumerGroup, instance);
+        }
+        String addrTopicGroupKey = RocketMqSubscriptionDataUtils.buildAddrTopicGroupKey(subscriptionData.getTopic(),
+                consumerGroup, namesrvAddr);
+        RocketMqSubscriptionDataUtils.resetsSql92SubscriptionData(subscriptionData, addrTopicGroupKey);
```

Comment:
Extract as a method, explain the function of the code by the method name.

Reply:
What content is extracted as a method

Reply:
fixed